business_rules/operators.py

```diff
@@ -782,6 +782,37 @@ def detect_reference(self, row, value_column, target_column, context=None):
         value = row[value_column]
         return (value in target_data) or (value in target_data.astype(int).astype(str)) or (value in target_data.astype(str))
 
+    @type_operator(FIELD_DATAFRAME)
+    def additional_columns_empty(self, other_value: dict):
+        """
+        The dataframe column might have some additional columns.
+        If the next additional column exists, the previous one cannot be empty.
+        Example:
+            column - TSVAL
+            additional columns - TSVAL1, TSVAL2, ...
+            If TSVAL2 exists -> TSVAL1 cannot be empty.
+            Original column (TSVAL) can be empty.
+
+        The operator extracts these additional columns from the DF
+        and ensures they are not empty.
+        """
+        target: str = self.replace_prefix(other_value.get("target"))
+        regex: str = rf"^{target}\d+$"  # starting from target, ending with integers and nothing is between them
+        df: pd.DataFrame = self.value.filter(regex=regex)
+        # applying a function to each row
+        result: pd.Series = df.apply(lambda row: self.next_column_exists_and_previous_is_null(row), axis=1)
+        return result
+
+    @type_operator(FIELD_DATAFRAME)
+    def additional_columns_not_empty(self, other_value: dict):
+        return ~self.additional_columns_empty(other_value)
+
+    def next_column_exists_and_previous_is_null(self, row: pd.Series) -> bool:
+        for index in range(len(row) - 1):
+            if row[index] is None and row[index + 1] is not None:
+                return True
+        return False
```

Comment:
I don't like the loops when working with DataFrames but here the row should not contain a lot of items. The row holds values of TSVAL1 - TSVALn, I do not think that we can have ...TSVAL3000 in dataset. So, O(n) complexity should still work very fast here.

On the other hand, the dataset can have thousands of rows but since pandas parallels apply() calls (line 803), having an O(n) algorithm being parallelized is fine.